controllers/etcd_controller.go

```diff
@@ -410,87 +411,6 @@ func (r *EtcdReconciler) delete(ctx context.Context, etcd *druidv1alpha1.Etcd) (
 	return ctrl.Result{}, nil
 }
 
-func (r *EtcdReconciler) reconcileServices(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd, renderedChart *chartrenderer.RenderedChart) (*corev1.Service, error) {
-	logger.Info("Reconciling etcd services")
-
-	selector, err := metav1.LabelSelectorAsSelector(etcd.Spec.Selector)
-	if err != nil {
-		logger.Error(err, "Error converting etcd selector to selector")
-		return nil, err
-	}
-
-	// list all services to include the services that don't match the etcd`s selector
-	// anymore but has the stale controller ref.
-	services := &corev1.ServiceList{}
-	err = r.List(ctx, services, client.InNamespace(etcd.Namespace), client.MatchingLabelsSelector{Selector: selector})
-	if err != nil {
-		logger.Error(err, "Error listing services")
-		return nil, err
-	}
-
-	// NOTE: filteredStatefulSets are pointing to deepcopies of the cache, but this could change in the future.
-	// Ref: https://github.com/kubernetes-sigs/controller-runtime/blob/release-0.2/pkg/cache/internal/cache_reader.go#L74
-	// if you need to modify them, you need to copy it first.
-	filteredServices, err := r.claimServices(ctx, etcd, selector, services)
-	if err != nil {
-		logger.Error(err, "Error claiming service")
-		return nil, err
-	}
-
-	if len(filteredServices) > 0 {
-		logger.Info("Claiming existing etcd services")
-
-		// Keep only 1 Service. Delete the rest
-		for i := 1; i < len(filteredServices); i++ {
-			ss := filteredServices[i]
-			if err := r.Delete(ctx, ss); err != nil {
-				logger.Error(err, "Error in deleting duplicate StatefulSet")
-				continue
-			}
-		}
-
-		// Return the updated Service
-		service := &corev1.Service{}
-		err = r.Get(ctx, types.NamespacedName{Name: filteredServices[0].Name, Namespace: filteredServices[0].Namespace}, service)
-		if err != nil {
-			return nil, err
-		}
-
-		// Service is claimed by for this etcd. Just sync the specs
-		if service, err = r.syncServiceSpec(ctx, logger, service, etcd, renderedChart); err != nil {
-			return nil, err
-		}
-
-		return service, err
-	}
-
-	// Required Service doesn't exist. Create new
-
-	svc, err := r.getServiceFromEtcd(etcd, renderedChart)
-	if err != nil {
-		return nil, err
-	}
-
-	err = r.Create(ctx, svc)
-
-	// Ignore the precondition violated error, this service is already updated
-	// with the desired label.
-	if err == errorsutil.ErrPreconditionViolated {
-		logger.Info("Service precondition doesn't hold, skip updating it.", "service", kutil.Key(svc.Namespace, svc.Name).String())
-		err = nil
-	}
-
-	if err != nil {
-		return nil, err
-	}
-
-	if err := controllerutil.SetControllerReference(etcd, svc, r.Scheme); err != nil {
-		return nil, err
-	}
-
-	return svc.DeepCopy(), err
-}
-
 func (r *EtcdReconciler) syncServiceSpec(ctx context.Context, logger logr.Logger, svc *corev1.Service, etcd *druidv1alpha1.Etcd, renderedChart *chartrenderer.RenderedChart) (*corev1.Service, error) {
```

Comment:
We don't need this function anymore, right?

Reply:
Nice catch 👍  Removed in a895b0c6d0eec7a9e7d4fee2bf230c1fd8e20cba.